akka-stream-tests/src/test/scala/akka/stream/scaladsl/AcknowledgeSinkSpec.scala

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.ActorMaterializer
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.{ AkkaSpec, _ }
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+import scala.util.control.NoStackTrace
+
+class AcknowledgeSinkSpec extends AkkaSpec {
+  implicit val mat = ActorMaterializer()
+  val ex = new RuntimeException with NoStackTrace
+
+  def assertFuture(value: Any, fb: Future[Option[Any]]): Unit =
+    Await.result(fb, 200.millis) should be(Some(value))
+
+  def assertComplete(fb: Future[Option[Any]]): Unit =
+    Await.result(fb, 200.millis) should be(None)
+
+  "An AcknowledgeSink" must {
```

Comment:
I am a bit confused by the error scenarios in this test suite. Can you explain them a bit?